bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Display.java

```diff
@@ -138,12 +139,8 @@ public class Display extends Device {
 
 	/* Widget Table */
 	int freeSlot;
```

Comment:
- freeSlot should be unused now
- SWT is single threaded, why do we need ConcurrentHashMap here?
- I wonder if the handle / control is really 1:1 relationship. At least in the table it was possible to map 1:N. Have you tried to run all SWT tests **locally** on Windows with the new code?

Reply:
I ran test locally and eclipse ide still starts normally.
It would still be possible to add multiple handles per control.  The getControl() signature however clearly renders multiple controls per handle as not supported.